pkg/kclient/pods_test.go

```diff
@@ -0,0 +1,85 @@
+package kclient
+
+import (
+	"fmt"
+	"testing"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"k8s.io/apimachinery/pkg/watch"
+
+	ktesting "k8s.io/client-go/testing"
+)
+
+func TestWaitAndGetPod(t *testing.T) {
+
+	tests := []struct {
+		name    string
+		podName string
+		status  corev1.PodPhase
+		wantErr bool
+	}{
+		{
+			name:    "phase: running",
+			podName: "ruby",
+			status:  corev1.PodRunning,
+			wantErr: false,
+		},
+
+		{
+			name:    "phase: failed",
+			podName: "ruby",
+			status:  corev1.PodFailed,
+			wantErr: true,
+		},
+
+		{
+			name: "phase:	unknown",
+			podName: "ruby",
+			status:  corev1.PodUnknown,
+			wantErr: true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+
+			fkclient, fkclientset := FakeNew()
+			fkWatch := watch.NewFake()
+
+			// Change the status
+			go func() {
+				fkWatch.Modify(fakePodStatus(tt.status, tt.podName))
+			}()
+
+			fkclientset.Kubernetes.PrependWatchReactor("pods", func(action ktesting.Action) (handled bool, ret watch.Interface, err error) {
+				return true, fkWatch, nil
+			})
+
+			podSelector := fmt.Sprintf("deployment=%s", tt.podName)
+			timeout := int64(10)
+			watchOptions := metav1.ListOptions{
+				LabelSelector:  podSelector,
+				TimeoutSeconds: &timeout,
+			}
+			pod, err := fkclient.WaitAndGetPod(watchOptions, corev1.PodRunning, "Waiting for component to start")
+
+			if !tt.wantErr == (err != nil) {
+				t.Errorf(" client.WaitAndGetPod(string) unexpected error %v, wantErr %v", err, tt.wantErr)
+				return
```

Comment:
We can do `t.Fatal` if we want the execution to stop at this point. I mean instead of doing `t.Errorf` and `return`, we could use just `t.Fatal`.

Reply:
Ah, good to know about `t.Fatal`, I didn't realize the testing package had that too. I've updated the test to use it.